### PR TITLE
feat: add adjust vault view with transition

### PR DESCRIPTION
**Pull request by samsiegart**
refs #15 

(The gif looks choppy, but you can see the animations for yourself in the ipfs deployments without a local chain running)
Recording 2023-02-05 at 11 56 52

### CODE REVIEW

File: src/views/Vaults.tsx [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+import ReactViewSlider from 'react-view-slider';
+import type { VaultKey } from 'store/vaults';
 
-type PathDescriptionProps = { mode: ViewMode };
+type PathDescriptionProps = { mode: ViewMode; adjustVaultKey: VaultKey | null };
 
-const PathDescription = ({ mode }: PathDescriptionProps) => {
-  if (mode === ViewMode.Create) {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ViewSlider = ReactViewSlider as unknown as FunctionComponent<any>;
```
Comment (samsiegart) [inline src/views/Vaults.tsx:21]: This package doesn't export types so this is the only way I found to get ts to quiet down.

Comment (turadg) [reply src/views/Vaults.tsx:21]: Ah. Consider explaining that in a code comment.

I looked at the package and it had Flow types so we could convert to TS and publish a DefinitelyTyped definition but prolly not worth it.

File: src/views/Vaults.tsx [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+
+const PathDescription = ({ mode, adjustVaultKey }: PathDescriptionProps) => {
+  if (mode === ViewMode.Edit && adjustVaultKey) {
```
Comment (turadg) [inline src/views/Vaults.tsx:24]: This shares the first condition with the next if. Please move the check for the second condition within that block.

File: src/vite-env.d.ts [hunk 3]
```diff
@@ -46,3 +46,7 @@ declare module '@agoric/inter-protocol/src/interest-math' {
     currentCompoundedInterest: Ratio,
   ) => Amount<'nat'>;
 }
+
+declare module 'react-view-slider' {
+  export const ViewSlider;
```
Comment (turadg) [inline src/vite-env.d.ts:51]: I think if this described the type being exported you wouldn't need the ts-expect-error cast when you import it

**Review by turadg — APPROVED** [review]

File: src/vite-env.d.ts (hunk 3 above)
Comment (samsiegart) [reply src/vite-env.d.ts:51]: Unfortunately it doesn't seem to be respecting it. I get `[JSX element type 'ViewSlider' does not have any construct or call signatures](https://github.com/s-yadav/react-number-format/issues/290#top)` unless I cast it here.

File: src/views/Vaults.tsx (hunk 2 above)
Comment (samsiegart) [reply src/views/Vaults.tsx]: Done

File: src/views/Vaults.tsx (hunk 1 above)
Comment (samsiegart) [reply src/views/Vaults.tsx:23]: Added a comment

**Pull request merged**